### PR TITLE
add apiv2 healthcheck code

### PR DESCRIPTION
**Pull request by baude**
reworking binding and endpoint to actually work.  added documentation in swagger for and various return code possibilities.  add a good start on tests though we need some other container functions not yet implemented for that.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: pkg/api/server/register_healthcheck.go [hunk 1]
```diff
@@ -8,6 +8,29 @@ import (
...
+	// responses:
+	//   200:
+	//     $ref: "#/responses/HealthcheckRun"
+	//   404:
+	//     $ref: "#/responses/NoSuchContainer"
+	//   409:
+	//     description: container has no healthcheck or is not running
+	//   500:
+	//     $ref: '#/responses/InternalError'
 	r.Handle(VersionedPath("/libpod/containers/{name}/runhealthcheck"), s.APIHandler(libpod.RunHealthCheck)).Methods(http.MethodGet)
```
Comment (edsantiago) [inline pkg/api/server/register_healthcheck.go]: Should this `{name}` be changed to `{name:.*}` ?

File: pkg/api/server/register_healthcheck.go [hunk 2]
```diff
@@ -8,6 +8,29 @@ import (
 )
 
 func (s *APIServer) registerHealthCheckHandlers(r *mux.Router) error {
-	r.Handle(VersionedPath("/libpod/containers/{name}/runhealthcheck"), s.APIHandler(libpod.RunHealthCheck)).Methods(http.MethodGet)
+	// swagger:operation GET /libpod/containers/{name:.*}/runhealthcheck libpod libpodRunHealthCheck
```
Comment (jwhonce) [inline pkg/api/server/register_healthcheck.go]: `/libpod/containers/{name:.*}/runhealthcheck` should be `/libpod/containers/{name:.*}/healthcheck`.  The verb GET is good.

**Comment by baude** [comment]
rebased hoping to pass tests

**Pull request merged**